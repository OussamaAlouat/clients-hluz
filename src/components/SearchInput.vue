<!-- eslint-disable vuejs-accessibility/form-control-has-label -->
<template>
  <div  class="input-container">
    <div style="width:80%">
      <v-text-field
        placeholder="Set cups"
        outlined
        v-model="cupsValue"
        class="input-container__input-hluz"
        type="text"
        id="cups"
      >
        <template v-slot:append>
          <v-hover v-slot="{ hover }">
            <v-icon
              large
              :color="hover ? 'rgba(229, 0, 126, 255)' : ''"
              :disabled="!cupsValue" @click="searchCUPS()"
            >
              mdi-magnify
            </v-icon>
          </v-hover>
        </template>
      </v-text-field>
    </div>

  </div>
</template>

<script>
export default {
  name: 'SearchInput',
  data: () => ({
    cupsValue: '',
  }),

  methods: {
    searchCUPS() {
      if (this.cupsValue) {
        this.$emit('cupsToSearch', this.cupsValue);
      }
    },
  },
};
</script>

<style lang="scss">
.input-container {
  display: flex;
  justify-content: center;
  margin-top: 20px;
  margin-bottom: 20px;

  &__input-hluz {
    border-radius: 10px;
    height: 60px;
    justify-content: center;
  }
}
</style>
