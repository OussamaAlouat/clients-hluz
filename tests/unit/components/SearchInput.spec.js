import { shallowMount } from '@vue/test-utils';
import SearchInput from '@/components/SearchInput.vue';

describe('SearchInput.vue', () => {
  it('Component render', () => {
    const wrapper = shallowMount(SearchInput, {});
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('On click on searchCUPS function will emit "cupsToSearch" event', () => {
    const wrapper = shallowMount(SearchInput, {});
    wrapper.setData({
      cupsValue: 'cupsValue',
    });

    wrapper.vm.searchCUPS();
    expect(wrapper.emitted()).toHaveProperty('cupsToSearch');
  })
});
