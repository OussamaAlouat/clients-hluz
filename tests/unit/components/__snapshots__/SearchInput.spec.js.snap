// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput.vue Component render 1`] = `
<div class="input-container">
  <div style="width: 80%;">
    <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" backgroundcolor="" id="cups" loaderheight="2" clearicon="$clear" outlined="true" placeholder="Set cups" type="text" class="input-container__input-hluz"></v-text-field-stub>
  </div>
</div>
`;
